lib/Sema/SemaBounds.cpp

```diff
@@ -374,12 +392,26 @@ namespace {
     }
 
     // Compute bounds for the target of an lvalue.  Values assigned through
-    // the lvalue must meet satisfy these bounds.   Values read through the
+    // the lvalue must satisfy these bounds.   Values read through the
     // lvalue will meet these bounds.
     BoundsExpr *LValueTargetBounds(Expr *E) {
       assert(E->isLValue());
       // TODO: handle side effects within E
       E = E->IgnoreParens();
+      QualType QT = E->getType();
+
+      // If the target value v is a Ptr type, it has bounds(v, v + 1), unless
+      // it is a function pointer type, in which case it has no required
+      // bounds.
+      if (QT->isCheckedPointerPtrType()) {
+         if (QT->isFunctionPointerType())
+           return CreateBoundsNone();
```

Comment:
We only allow ptrs to functions in the semantics.   Array_ptrs to functions are not allowed.    A bound for a pointer to a function makes no sense in C because the size of a function is unknown.  If a f is a function pointer, f + 1 has no meaning.   On the other hand, you can have a bound for a pointer to a pointer to a function.